### PR TITLE
refactor(j-s): use boolean selectors in formatmessage

### PR DESCRIPTION
**Pull request by Arnarkari93**
[Asana](https://app.asana.com/0/1199153462262248/1202211680145828/f)

## What

- Send boolean to formatMessage

## Why

- We need to create a new version of the strings becuase the current
translations for these strings are wrong.
- So that we don't have examples in our code using `yes` or `no` string provided to the formatmessage

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/judicial-system/backend/src/app/formatters/formatters.ts [hunk 1]
```diff
@@ -309,19 +309,19 @@ export function formatPrisonCourtDateEmailNotification(
...
     sessionArrangements,
   })
 
-  return formatMessage(notifications.prisonCourtDateEmail.body, {
+  return formatMessage(notifications.prisonCourtDateEmail.bodyV2, {
     caseType: type,
     prosecutorOffice: prosecutorOffice || 'NONE',
     courtText,
-    isExtension: isExtension ? 'yes' : 'no',
+    isExtension: isExtension,
```
Comment (oddsson) [inline apps/judicial-system/backend/src/app/formatters/formatters.ts:324]:
```suggestion
    isExtension,
```

File: apps/judicial-system/backend/src/app/formatters/formatters.ts [hunk 2]
```diff
@@ -501,10 +501,10 @@ export function formatPrisonRevokedEmailNotification(
   const defenderText = formatMessage(cf.defender, {
     defenderName: defenderName || 'NONE',
   })
-  const revokedCaseText = formatMessage(cf.revokedCase, {
+  const revokedCaseText = formatMessage(cf.revokedCaseV2, {
     caseType: type,
     prosecutorOffice: prosecutorOffice || 'NONE',
-    isExtension: isExtension ? 'yes' : 'no',
+    isExtension: isExtension,
```
Comment (oddsson) [inline apps/judicial-system/backend/src/app/formatters/formatters.ts:507]:
```suggestion
    isExtension,
```

File: apps/judicial-system/web/src/routes/Court/RestrictionCase/Ruling/Ruling.tsx [hunk 3]
```diff
@@ -123,23 +119,16 @@ export function getConclusionAutofill(
...
           ?.replace(' kl.', ', kl.')}`,
-        hasIsolation:
-          isAcceptingCaseDecision(decision) && isCustodyIsolation
-            ? 'yes'
-            : 'no',
-        isolationEndsBeforeValidToDate: isolationEndsBeforeValidToDate
-          ? 'yes'
-          : 'no',
+        hasIsolation: isAcceptingCaseDecision(decision) && isCustodyIsolation,
+        isolationEndsBeforeValidToDate: isolationEndsBeforeValidToDate,
```
Comment (oddsson) [inline apps/judicial-system/web/src/routes/Court/RestrictionCase/Ruling/Ruling.tsx:131]:
```suggestion
        isolationEndsBeforeValidToDate,
```

**Review by oddsson — CHANGES_REQUESTED** [review]
I had a couple of suggestions for changes. Also, there is a warning about `renderSuccessText` not being used. We should probably look into that before merging.

File: apps/judicial-system/web/src/routes/Court/RestrictionCase/Ruling/Ruling.tsx (hunk 3 above)
Comment (Arnarkari93) [reply apps/judicial-system/web/src/routes/Court/RestrictionCase/Ruling/Ruling.tsx]: [b9da764](https://github.com/island-is/island.is/pull/7894/commits/b9da76477094fe1622451f8238ea898e741399b7)

**Comment by Arnarkari93** [comment]
> I had a couple of suggestions for changes. Also, there is a warning about `renderSuccessText` not being used. We should probably look into that before merging.

Addressed here: [335c528](https://github.com/island-is/island.is/pull/7894/commits/335c528043604c93e1c4a5d96e9c9f4f0a64505c) there is also a function `getSuccessText` that serves the same purpose. I just forgot to remove the old one.

File: apps/judicial-system/backend/src/app/formatters/formatters.ts (hunk 1 above)
Comment (oddsson) [inline apps/judicial-system/backend/src/app/formatters/formatters.ts:324]:
```suggestion
    isExtension,
```

**Review by oddsson — CHANGES_REQUESTED** [review]
One more thing ;)

**Review by oddsson — APPROVED** [review]

**Pull request merged**